From 458f73561247747f6dc0ab14542c55864be0b421 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Fri, 16 Jan 2015 19:25:26 +0100
Subject: [PATCH 084/102] sparsify: handle output filenames with ':'

With commit a594b7f90a682e2a9327b142138edb76ad8ba8ff the checks for
input and output files with ':' were removed; while the input file is
handled correctly by the disk_create API, an output filename with ':'
could still be handled as specifying a transport, failing the last
"qemu-img convert" call.

Use the new qemu_input_filename helper to handle the output filename
correctly.

(cherry picked from commit 1709263613ac97cb1f37333e1a429c34f6b79f26)
---
 sparsify/sparsify.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sparsify/sparsify.ml b/sparsify/sparsify.ml
index d6b1193..84d5a5f 100644
--- a/sparsify/sparsify.ml
+++ b/sparsify/sparsify.ml
@@ -491,7 +491,7 @@ let () =
       (match option with
       | None -> ""
       | Some option -> " -o " ^ Filename.quote option)
-      (Filename.quote overlaydisk) (Filename.quote outdisk) in
+      (Filename.quote overlaydisk) (Filename.quote (qemu_input_filename outdisk)) in
   if verbose then
     printf "%s\n%!" cmd;
   if Sys.command cmd <> 0 then
-- 
1.9.3

